### PR TITLE
`Development`: Refactor quiz exercise detail page

### PR DESCRIPTION
**Pull request by rriyaldhi**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The implementation of the Quiz Pool Configuration Page will reuse the functionalities of the Quiz Exercise Detail Page. Therefore, The Quiz Exercise Detail Page is refactored so the functionality can be reused later by the Quiz Pool Configuration page.

### Description

1. Move the UI of the list of Quiz Question Cards to a separate component: QuizQuestionListEditComponent
The list of Quiz Question Cards will also be used on Quiz Pool Configuration Page. Therefore, the UI elements are refactored to a new reusable component.
2. Move the UI for importing existing Quiz Questions to a separate component: QuizQuestionListEditExistingComponent
3. Move the checking of Quiz Questions (if it is valid or not) to the isQuizQuestionValid method.
The checking of Quiz Questions will also be used by the Quiz Pool Configuration Page, so the logic is moved to a separate reusable method.
5. Add ViewEncapsulation.None to MultipleChoiceQuestionEditComponent to make the style consistent with DragAndDropQuestionEditComponent and ShortAnswerQuestionEditComponent.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
6. Select the Course
7. Navigate to Exercises
8. Create a new Quiz Exercise
9. Verify that adding new Quiz Questions works
10. Add new Quiz Questions, then Save
11. Edit the newly created Quiz Exercise
12. Verify that the Quiz Questions are displayed correctly
13. Import existing Quiz Questions from either another course, exam, or JSON file.
14. Save the exercise
15. Verify that the imported Quiz Questions are saved

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| multiple-choice-question-edit.component.ts | 87.5% | 95.06% |
| quiz-exercise-detail.component.ts | 76.49% | 89.88% |
| quiz-exercise-validation.directive.ts | 81.96% | 93.22% |
| quiz-question-list-edit-existing.component.ts | 80.55% | 96.7% |
| quiz-question-list-edit.component.ts | 100% | 97.05% |
| quiz-manage-util.service.ts | 96.21% | 99.04% |

### Screenshots

https://user-images.githubusercontent.com/10885503/230735986-53187e78-aac8-4552-a872-a42fa7c6d3cf.mov

### CODE REVIEW

**Review by DominikRemo — APPROVED** [review]
Tested on ts2 in testing session. Works as expected

**Review by danielbode — DISMISSED** [review]
Works as expected. Tested on ts2 during a testing session.

**Review by YeetTheFirst21 — DISMISSED** [review]
Tested on TS2 in the testing session, all seems to be working properly.

**Review by laadvo — DISMISSED** [review]
Tested during the testing session on ts2. Everything working as expected

**Review by chrisknedl — APPROVED** [review]
Tested on TS2. Everything works as intended.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.html
Comment (frederik-kt) [inline src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.html:213]: How can you be certain that `courseId` and `quizExercise.quizQuestions` are never `null` or `undefined`? I think it would be better to be defensive and check for `null` or `undefined` in the component that accesses them. The default value in `@Input() quizQuestions: QuizQuestion[] = [];` in the `quiz-question-list-edit.component` will be overwritten if the parent component sets `null` or `undefined` as the input.

Comment (rriyaldhi) [reply src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.html]: Thanks for noticing! Added a wrapper with ngIf of jhi-quiz-question-list-edit component. Also added a check if quizQuestions is undefined.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts
Comment (rriyaldhi) [reply src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts:50]: Unused fields are removed now.

**Review by frederik-kt — DISMISSED** [review]
Code LGTM

**Review by TheZoker — APPROVED** [review]
Tested within testing session on TS3 and LGTM 👍

**Review by FelixTJDietrich — DISMISSED** [review]
Tested during testing session on ts3, works as expected.

**Review by pellumbbaboci — APPROVED** [review]
tested on ts3 during testing session and everything works

**Review by Santia-go — DISMISSED** [review]
Tested on TS1
The creation of a quiz works as expected.

All parameters are inserted and visualized correctly.
All kind of questions works as expected.
The resolving process and the assessment are working correctly.
The edition, reevaluation and reset of a quiz are working.
Deleting and importing are also working.
The same was tested in an exam with two quizzes.

Importing the quiz to an exam works and importing questions into a quiz also works
Presenting the summary of the exam also works
The assessment of the quiz works as expected

**Review by YeetTheFirst21 — DISMISSED** [review]
Tested on TS1, all seems to work as expected.

Also checked some edge cases like deleting + editing quiz while it was running with no issues

**Review by frederik-kt — APPROVED** [review]
Latest changes look good to me.

**Review by YeetTheFirst21 — APPROVED** [review]
Reapprove after conflict fix

**Review by Santia-go — APPROVED** [review]
Re-approved after fixing the merge conflict

**Review by FelixTJDietrich — APPROVED** [review]
Re-approve after merge conflict fixes

**Review by greekgeek1 — APPROVED** [review]

**Pull request merged**
